### PR TITLE
OAST API Support

### PR DESCRIPTION
**Pull request by psiinon**
## Overview
OAST API Support.
WIP pending more endpoints 😁 

## Checklist
- [ ] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/oast/CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
-
+### Added API support.
```
Comment (kingthorin) [inline addOns/oast/CHANGELOG.md:9]: API support as a bullet?

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/OastAPI.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import org.zaproxy.zap.extension.api.ApiAction;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+import org.zaproxy.zap.extension.api.ApiResponseList;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+import org.zaproxy.zap.extension.api.ApiView;
+
+public class OastAPI extends ApiImplementor {
```
Comment (kingthorin) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/OastAPI.java:33]: OastApi

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/OastAPI.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                ext.getCallbackService()
+                        .getParam()
+                        .setLocalAddress(params.getString(PARAM_LOCAL_ADDR));
+                ext.getCallbackService()
+                        .getParam()
+                        .setRemoteAddress(params.getString(PARAM_REMOTE_ADDR));
+                try {
+                    ext.getCallbackService()
+                            .getParam()
+                            .setPort(Integer.parseInt(params.getString(PARAM_PORT)));
```
Comment (kingthorin) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/OastAPI.java:118]: Don't we have a param getter for ints?

File: addOns/oast/CHANGELOG.md (hunk 1 above)
Comment (psiinon) [reply addOns/oast/CHANGELOG.md:9]: Arg, I've got that wrong before :P

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/OastAPI.java (hunk 2 above)
Comment (psiinon) [reply addOns/oast/src/main/java/org/zaproxy/addon/oast/OastAPI.java:33]: We seem to have a mix of API and Api, but I'm easy, so I'll change it 😁

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            case ACTION_SET_INTERACTSH_OPTIONS:
+                ext.getInteractshService().getParam().setServerUrl(params.getString(PARAM_SERVER));
+                ext.getInteractshService()
+                        .getParam()
+                        .setAuthToken(params.getString(PARAM_AUTH_TOKEN));
+                try {
+                    ext.getInteractshService()
+                            .getParam()
+                            .setPollingFrequency(
+                                    Integer.parseInt(params.getString(PARAM_POLL_IN_SECS)));
```
Comment (kingthorin) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java:131]: params.getInt

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                            ApiException.Type.ILLEGAL_PARAMETER, params.getString(PARAM_NAME));
+                }
+                break;
+            case ACTION_SET_BOAST_OPTIONS:
+                ext.getBoastService().getParam().setBoastUri(params.getString(PARAM_SERVER));
+                try {
+                    ext.getBoastService()
+                            .getParam()
+                            .setPollingFrequency(
+                                    Integer.parseInt(params.getString(PARAM_POLL_IN_SECS)));
```
Comment (kingthorin) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java:101]: params.getInt

**Comment by psiinon** [comment]
Ready to review.
More changes planned, but these should be a coherent set of changes that can be reviewed on their own more easily...

**Review by kingthorin — COMMENTED** [review]
Add an API page to the help?

Other than that it looks good to me.

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java (hunk 4 above)
Comment (thc202) [reply addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java]: It's better to use e.g. `ApiUtils.getIntParam(params, PARAM_POLL_IN_SECS)` which already handles the possible missing/invalid values.

**Comment by psiinon** [comment]
Good point, looks like quite a few add-ons have an API page, so I'll add one

**Comment by psiinon** [comment]
Updated based on feedback and ready to review again..

**Review by kingthorin — APPROVED** [review]
LGTM

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+import org.zaproxy.zap.extension.api.ApiResponseList;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+import org.zaproxy.zap.extension.api.ApiView;
+import org.zaproxy.zap.utils.ApiUtils;
+
+public class OastApi extends ApiImplementor {
+    private static final String PREFIX = "oast";
+
+    private static final String ACTION_SET_ACTIVE_SERVICE = "setActiveService";
```
Comment (ricekot) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java:37]: Could be called `setActiveScanService` for clarity

**Review by ricekot — APPROVED** [review]
Thanks!

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java (hunk 6 above)
Comment (psiinon) [reply addOns/oast/src/main/java/org/zaproxy/addon/oast/OastApi.java]: Names changed :)

**Pull request merged**